modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/parser/java/JavaExtensionModelParserUtils.java

```diff
@@ -93,7 +93,8 @@ public static List<ExtensionParameter> getCompletionCallbackParameters(MethodEle
         .filter(p -> {
           Type type = p.getType();
           return type.isAssignableTo(CompletionCallback.class) ||
-              type.isAssignableTo(org.mule.sdk.api.runtime.process.CompletionCallback.class);
+              type.isAssignableTo(org.mule.sdk.api.runtime.process.CompletionCallback.class) ||
+              type.isAssignableTo(org.mule.sdk.api.runtime.process.VoidCompletionCallback.class);
```

Comment:
Not exactly, that one considers RouterCompletionCallback, and here it's completion callback (so more cases). Also adding here RouterCompletionCallback is not necessary since it's already considered by CompletionCallback